usr/src/uts/common/fs/zfs/metaslab.c

```diff
@@ -1477,10 +1538,17 @@ metaslab_fini(metaslab_t *msp)
 		range_tree_destroy(msp->ms_defertree[t]);
 	}
 
+	metaslab_free_trimset(msp->ms_cur_ts);
```

Comment:
The 7303 patch in 8363e80a would cause in-flight trim zio to fail.  I added a patch to make sure the trimming was finshed (earlier in this function - review comment is here because it can't be added to an un-changed code area).  See dweeezil/zfs@62345698  for details.